src/proxy.c

```diff
@@ -56,6 +56,12 @@ static SEXP vec_proxy_unwrap(SEXP x) {
   return x;
 }
 
+// [[ register() ]]
+SEXP vctrs_unset_s4(SEXP x) {
```

Comment:
Can you make this `r_unmark_s4()` in utils.c please. The mark/unmark terminology conveys that the object is modified in place (like poke). Set and unset functions should make a copy if necessary.

Then add a `r_unset_s4()` variant that makes a copy.